src/collect/collector.rs

```diff
@@ -115,6 +115,7 @@ impl Collectors {
     }
 
     // Register the dynamic commands with the cli and parse collector-specific arguments
+    #[allow(dead_code)]
     fn register_cli(&mut self, mut cli: FullCli) -> Result<CliConfig> {
```

Comment:
Is this used anywhere? IMO looks like the function could be removed instead.

Reply:
> Is this used anywhere now? IMO looks like the function could be removed instead.

Yes, we still need it. e.g.register the collectors in full()
```rust
        let collectors = modules.collectors();
        let _ = collectors
            .values()
            .try_for_each(|c| c.register_cli(&mut self.collectors));
```
Or modules registers, e.g.

```rust
impl Collector for OvsModule {

    fn register_cli(&self, cmd: &mut DynamicCommand) -> Result<()> {
        cmd.register_module::<OvsCollectorArgs>(ModuleId::Ovs)
    }

```

Reply:
But the above example is the `Collector` trait implementation, for actual collectors. The function I'm suggesting to remove is an implementation of the `Collectors` command, not a trait implementation. Things are a little bit confusing here, I admit :)

Reply:
Ah, got it. I will update the patch.